File: ref/client/Assets/uLua/Lua/Custom/Common/TipWithTween.lua
TipWithTween = {
	uiName = 'TipWithTween',
	message = "",

}


function TipWithTween:Awake()
	self.uiMessage = self.transform:FindChild("Info"):GetComponent("UILabel")
end

function TipWithTween:Start()
	-- GameObject:Destroy(self.transform.gameObject, 2)
end

function TipWithTween:SetMessage(message)
	self.message = message
	self.uiMessage.text = self.message
	GameObject.Destroy(self.transform.gameObject, 1)
end

return TipWithTween

File: ref/client/Assets/Code/Game/GUI/TipWithTween.cs
using UnityEngine;
using System.Collections;

public class TipWithTween : MonoBehaviour
{
    private UILabel _title;
    private UILabel _message;

    public string message
    {
        get { return _message.text; }
        set { _message.text = value; }
    }
    void Awake()
    {
        _title = transform.FindChild("Title").GetComponent<UILabel>();
        _message = transform.FindChild("Info").GetComponent<UILabel>();
        _message.text = CommonFunction.GetConstString("STORE_BUY_GOODS_GOLD_NOT_ENOUGH");
        GetComponent<TweenAlpha>().ResetToBeginning();
        GetComponent<TweenAlpha>().PlayForward();
        GameObject.Destroy(gameObject, 2f);
    }
}
